extensions/amp-script/0.1/amp-script.js

```diff
@@ -454,9 +454,11 @@ export class AmpScriptService {
     this.sources_ = [];
 
     // Query the meta tag once per document.
-    const allowedHashes = ampdoc
-      .getHeadNode()
-      .querySelector('meta[name="amp-script-src"]');
+    // Note: <meta> tags are only valid in <head> elements, so regardless of
+    // the specific type of ampdoc, the top document head must be referenced.
+    // For example, there is no <head> element in an AmpDocShadow instance.
+    const headNode = dev().assertElement(ampdoc.win.document.head);
+    const allowedHashes = headNode.querySelector('meta[name="amp-script-src"]');
```

Comment:
1. Is this the best approach, or would it be better to specially handle just the AmpDocShadow case?
   ```
   let headNode = ampdoc.getHeadNode();
   if (headNode instanceof ShadowRoot) {
      headNode = ampdoc.win.document.head;
   }
   ```
2. There have been murmurings over time that _maybe_ someday multiple AMP documents could be displayed at once in a PWA. Should we prepare for this possibility? I see a couple of options:
   1. `querySelectorAll('meta[name="amp-script-src"]')`
   2. When `<meta name="amp-script-src" ...>` is imported into `<head>`, if there already exists the same type of `<meta>` tag, concatenate their `content` values

Reply:
@kevinkimball Can you take a look at this?

Reply:
@mdmower there is support for multiple documents shown on one PWA today - that is the use case for the multidoc-manager.js file you edited, so I think we do need to support multiple meta tags to move forward

Reply:
Ready for re-review. From my original review:
1. Opted to limit AmpScriptService changes to shadow docs
2. Opted for ii: combine multiple ampdoc meta tags into a single meta tag in owner doc